### DISCUSSION TITLE
trade-off and abundance

### DISCUSSION THREAD

**Discussion by oMersa344** [micom-dev/micom#55] [Q&A] [answered]
Hi,

I'm trying to model a simple community of 2 microbes and one yeast strain. the microbes individual models were constructed by CarveMe and the yeast model was downloaded from the BIGG database (iMM904 model for S288C). 
When I set the abundance of the yeast for 0.5 and 0.25 for each bacteria strain I get those growth rates with ```com.optimize()``` - 
```
compartments  abundance  growth_rate  reactions  metabolites                                               
S288c              0.50     2.311406       1577         1226
l_plantarum        0.25     0.763927       1462          950
l_sanfran          0.25     0.255284       1048          741
medium              NaN          NaN        298          298
```
However, when I try to implement a trade-off values set it's changing dramatically, and surprisingly the ratio between the growth rate is very similar to the abundance ratio (2:1:1), for all trade-off values. 
```
compartments     S288c  l_plantarum  l_sanfran  medium
0             1.886270     0.944876   0.924704     NaN
1             1.739854     0.849432   0.753317     NaN
2             1.510765     0.756318   0.736307     NaN
3             1.316010     0.658775   0.658738     NaN
4             1.126364     0.564719   0.566934     NaN
5             0.943098     0.472150   0.464755     NaN
6             0.757357     0.378899   0.368363     NaN
7             0.569126     0.285494   0.267892     NaN
8             0.376546     0.188362   0.191291     NaN
9             0.187560     0.093763   0.094560     NaN
```
When I change the abundance, the ratio of the growth rates is changing accordingly.

Is this expected? does a trade-off is actually necessary for this simple community model?

Thank you.

**Comment by cdiener** [comment, marked as answer]
Great question. Simulating small controlled co-cultures is indeed different than simulating a full (environmental) microbiome.

You are correct that in a rich medium and with few taxa we usually see that you can use a pretty large tradeoff value close to 1.0. This means the full community growth rate can be maintained without sacrificing any growth for individual taxa. However, there is still the issue of non-uniqueness. In your example, the individual growth rate sets of `[2.31, 0.76, 0.0.26]` and `[1.87, 0.94, 0.92]` both yield the same total biomass production, and there are infinitely many more that do as well. So which one is right? I would argue the first one is less likely since S288c grows extremely fast and the two closely related and same abundance taxa grow with very different growth rates. 

It's also true that in rich media the individual growth rates are correlated with abundance (there is a derivation in the paper). Usually, this is okay since in most environments, and during a steady state, the most abundant species is the fittest one (largest growth rate). For co-cultures, this can lead to slightly bad estimates if one uses the inoculation abundance and not the abundance during the exponential phase. If you want to simulate from inoculation abundances alone this would not be a steady-state model and I would use something like dFBA. This will require additional data like uptake kinetics though.

Whether the growth rates that are predicted are actually correct has to be validated with data in the end, but the growth rates returned by cooperative tradeoff even with a tradeoff parameter of 1.0 are at least consistent (same solution when re-running the optimization) and at least empirically match experimental data in various settings.

A bit unrelated, what medium do you use for the simulation? Did you use the same medium as used in CARVME for gapfilling?

**Reply by oMersa344** [reply]
Hi, thank you for the helpful answer.

As for your question - the medium is rich, supplemented with glucose and maltose as the carbon sources, aimed to mimic the medium I'm using at the lab. and yes, after creating the model at micom with ``` com = Community(data) ``` I applied the new medium with ``` com.medium = med ```.

Is there another method I can use to model the growth of co-cultures other than dFBA?
In more detail - I have 10 genomes of different LAB and I want to model all the possible pairs in different co-cultures and analyze which are the pairs that earn from it (e.g., the total biomass flux of the co-culture is larger than the biomass flux of each individual culture alone). Do you know a method to analyze it without the kinetics data dFBA requires?

Thank you

**Reply by cdiener** [reply]
You can use MICOM but the abundances will represent the log phase abundances rather than the inoculation. If you don't know those then using uniform abundances or at least the same abundances for each simulation is okay. This will be enough to predict  mutualism and competition. Inoculation ratios aren't optimal because the growth rates together with the inoculation will dictate the log-phase abundances. If you inoculate a small amount of a fast growing organism it will still reach high abundance in log-phase. If you want to model the full growth curve this is not a steady state system and that needs a dynamic model which will always require additional time-series data. There is unfortunately no free lunch there.

**Reply by oMersa344** [reply]
I'll try it out. As for now, I'm interested in the mutualism/competition analysis in steady-state and not the full growth curve, but maybe it should be the next thing on my mind. 
Thank you
